activation/nipost_test.go

```diff
@@ -807,6 +807,90 @@ func TestNIPSTBuilder_PoetUnstable(t *testing.T) {
 	})
 }
 
+func TestPoetTimeoutsParameters(t *testing.T) {
+	t.Parallel()
+
+	sig, err := signing.NewEdSigner()
+	require.NoError(t, err)
+
+	challenge := &types.NIPostChallenge{
+		PublishEpoch: postGenesisEpoch + 2,
+	}
+
+	t.Run("submit challenge timeout is too short -> fail", func(t *testing.T) {
+		db := localsql.InMemory()
+
+		require.NoError(t, nipost.AddChallenge(db, sig.NodeID(), challenge))
+		challengeHash := wire.NIPostChallengeToWireV1(challenge).Hash()
+
+		ctrl := gomock.NewController(t)
+		mclock := defaultLayerClockMock(ctrl)
+
+		poetProver := NewMockPoetService(ctrl)
+		poetProver.EXPECT().
+			Submit(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), sig.NodeID()).
+			Return(nil, context.DeadlineExceeded)
+		poetProver.EXPECT().Address().AnyTimes().Return("http://localhost:9999")
+
+		postService := NewMockpostService(ctrl)
+
+		nb, err := NewNIPostBuilder(
+			db,
+			postService,
+			zaptest.NewLogger(t),
+			PoetConfig{SubmitChallengeTimeout: 1 * time.Microsecond},
+			mclock,
+			nil,
+			WithPoetServices(poetProver),
+		)
+		require.NoError(t, err)
+
+		nipst, err := nb.BuildNIPost(context.Background(), sig, challengeHash, challenge)
+		require.ErrorIs(t, err, ErrPoetServiceUnstable)
+		require.Nil(t, nipst)
+	})
```

Comment:
This test doesn't actually test what the name of it would suggest. It rather tests that `BuildNIPost` returns `ErrPoetServiceUnstable` if `Submit` fails. This is covers the same code as `TestNIPSTBuilder_PoetUnstable/submit_fails`. Additionally `TestNIPSTBuilder_PoetUnstable/submit_hangs` actually tests that if the submit hangs and has to be canceled `BuildNiPost` does this and returns `ErrPoetServiceUnstable`.

Reply:
I removed both tests because there are no separate timeouts anymore